presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTile.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.spi.PrestoException;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public final class BingTile
+{
+    private final int x;
+    private final int y;
+    private final int zoomLevel;
+
+    BingTile(int x, int y, int zoomLevel)
+    {
+        this.x = x;
+        this.y = y;
+        this.zoomLevel = zoomLevel;
+    }
+
+    public int getX()
+    {
+        return x;
+    }
+
+    public int getY()
+    {
+        return y;
+    }
+
+    public int getZoomLevel()
+    {
+        return zoomLevel;
+    }
+
+    public static BingTile fromQuadKey(String quadKey)
+    {
+        int zoomLevel = quadKey.length();
+        int tileX = 0;
+        int tileY = 0;
+        for (int i = zoomLevel; i > 0; i--) {
+            int mask = 1 << (i - 1);
+            switch (quadKey.charAt(zoomLevel - i)) {
+                case '0':
+                    break;
+
```

Comment:
Remove the blank lines. The "break" statement provides ample vertical space for readability.